source/slang/lower-to-ir.cpp

```diff
@@ -3512,7 +3512,9 @@ static void lowerEntryPointToIR(
         // the entry point request.
         return;
     }
-
+    // we need to lower all global type arguments as well
+    for (auto arg : entryPointRequest->genericParameterTypes)
```

Comment:
Hmm... I was initially unclear on why this needs to be here. I was thinking that it *shouldn't* be, because `lower-to-ir` is supposed to generate the "initial" IR, which is independent of the chosen target, or the chosen type arguments for specialization, so even *looking* at the array of generic arguments here risks breaking things by making this pass not be independent.

But then I realized that maybe we have a problem in this pass where we are only generating initial IR for the declarations that are actually referenced by the entry point, so that we'd fail to output things like witness tables unless we did this step.

If that is the case, then this fix seem okay for now, but the right answer is really that we should have the `lower-to-ir` pass lower *everything*, and leave filtering to the stuff used by an entry point for the target-specific step.

Reply:
Yeah that is exactly what happened. The intial IR may not include the necessary witness tables for unused types, even if that type later get substituted in via a global generic parameter.